### PR TITLE
Closes #313: Read and convert ExtractedDocumentMetadata to AffMatchAffiliation

### LINKED ISSUE #313: Read and convert ExtractedDocumentMetadata to AffMatchAffiliation
Convert ExtractedDocumentMetadata to a newly created AffMatchAffiliation; normalize string values.

### PR DESCRIPTION
**Pull request by lukdumi**

### CODE REVIEW

**Comment by lukdumi** [comment]
A few notes:
AffMatchOrganization and AffMatchAffiliation will be changed to normal (not avro) classes in the near future. They are avro classes now just to check if the spark job reads/converts and writes affiliations and organizations correctly (just wanted to start to write a sensible job test, but now I think it hasn't been necessary).

File: iis-wf/iis-wf-affmatching/src/main/java/eu/dnetlib/iis/wf/affmatching/AffiliationConverter.java [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+*/
+
+public class AffiliationConverter {
+
+    
+    private StringNormalizer organizationNameNormalizer = new LenientComparisonStringNormalizer();
+    
+    private StringNormalizer countryNameNormalizer = new LenientComparisonStringNormalizer();
+    
+    private StringNormalizer countryCodeNormalizer = new LenientComparisonStringNormalizer();
```
Comment (mpol) [inline iis-wf/iis-wf-affmatching/src/main/java/eu/dnetlib/iis/wf/affmatching/AffiliationConverter.java]: Why AffMatchingJob uses static fields for utilities and this class doesn't?

File: iis-wf/iis-wf-affmatching/src/main/java/eu/dnetlib/iis/wf/affmatching/OrganizationConverter.java [hunk 2]
```diff
@@ -13,16 +17,22 @@
...
     
     private StringNormalizer organizationNameNormalizer = new LenientComparisonStringNormalizer();
     
+    private StringNormalizer organizationShortNameNormalizer = new LenientComparisonStringNormalizer();
+    
     private StringNormalizer countryNameNormalizer = new LenientComparisonStringNormalizer();
     
+    private StringNormalizer countryCodeNormalizer = new LenientComparisonStringNormalizer();
+    
     private StringNormalizer websiteUrlNormalizer = new WebsiteUrlNormalizer();
```
Comment (mpol) [inline iis-wf/iis-wf-affmatching/src/main/java/eu/dnetlib/iis/wf/affmatching/OrganizationConverter.java]: Same question: static like utilities in AffMatchingJob, or not?

File: iis-wf/iis-wf-affmatching/src/main/java/eu/dnetlib/iis/wf/affmatching/OrganizationConverter.java [hunk 3]
```diff
@@ -61,6 +74,10 @@ public void setOrganizationNameNormalizer(StringNormalizer organizationNameNorma
         this.organizationNameNormalizer = organizationNameNormalizer;
     }
 
+    public void setOrganizationShortNameNormalizer(StringNormalizer organizationShortNameNormalizer) {
+        this.organizationShortNameNormalizer = organizationShortNameNormalizer;
+    }
+
     public void setCountryNameNormalizer(StringNormalizer countryNameNormalizer) {
```
Comment (mpol) [inline iis-wf/iis-wf-affmatching/src/main/java/eu/dnetlib/iis/wf/affmatching/OrganizationConverter.java]: One setter missing. Or YAGNI?

File: iis-wf/iis-wf-affmatching/src/test/java/eu/dnetlib/iis/wf/affmatching/AffiliationConverterTest.java [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+        
+        // assert
+        
+        assertNotNull(affMatchAffiliations);
+        assertAffiliation(affMatchAffiliations.get(0), "XYZ", 1, "abc inst name", "Poland", "pl");
+        assertAffiliation(affMatchAffiliations.get(1), "XYZ", 2, "def inst name", "Germany", "de");
+        assertAffiliation(affMatchAffiliations.get(2), "XYZ", 3, "ghi inst name", "Slovakia", "sl");
+        
+    }
+
```
Comment (mpol) [inline iis-wf/iis-wf-affmatching/src/test/java/eu/dnetlib/iis/wf/affmatching/AffiliationConverterTest.java]: If two generated affiliations will get the same country code randomly - won't the test fail?

File: iis-wf/iis-wf-affmatching/src/main/java/eu/dnetlib/iis/wf/affmatching/OrganizationConverter.java (hunk 3 above)
Comment (lukdumi) [inline iis-wf/iis-wf-affmatching/src/main/java/eu/dnetlib/iis/wf/affmatching/OrganizationConverter.java]: :+1:

File: iis-wf/iis-wf-affmatching/src/test/java/eu/dnetlib/iis/wf/affmatching/AffiliationConverterTest.java (hunk 4 above)
Comment (lukdumi) [inline iis-wf/iis-wf-affmatching/src/test/java/eu/dnetlib/iis/wf/affmatching/AffiliationConverterTest.java]: :+1:

**Pull request closed without merging**